deployment/keystone/changeset/update_don.go

```diff
@@ -23,9 +47,73 @@ type UpdateDonResponse struct {
 // This a complex action in practice that involves registering missing capabilities, adding the nodes, and updating
 // the capabilities of the DON
 func UpdateDon(env deployment.Environment, req *UpdateDonRequest) (deployment.ChangesetOutput, error) {
-	_, err := internal.UpdateDon(env.Logger, req)
+	appendResult, err := AppendNodeCapabilities(env, appendRequest(req))
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to append node capabilities: %w", err)
+	}
+
+	ur, err := updateDonRequest(env, req)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to create update don request: %w", err)
+	}
+	updateResult, err := internal.UpdateDon(env.Logger, ur)
 	if err != nil {
 		return deployment.ChangesetOutput{}, fmt.Errorf("failed to update don: %w", err)
 	}
-	return deployment.ChangesetOutput{}, nil
+
+	out := deployment.ChangesetOutput{}
+	if req.UseMCMS() {
+		if updateResult.Ops == nil {
+			return out, fmt.Errorf("expected MCMS operation to be non-nil")
+		}
+		if len(appendResult.Proposals) == 0 {
+			return out, fmt.Errorf("expected append node capabilities to return proposals")
+		}
+
+		out.Proposals = appendResult.Proposals
+
+		// add the update don to the existing batch
+		// this makes the proposal all-or-nothing because all the operations are in the same batch, there is only one tr
+		// transaction and only one proposal
+		out.Proposals[0].Transactions[0].Batch = append(out.Proposals[0].Transactions[0].Batch, updateResult.Ops.Batch...)
```

Comment:
should we have validation to look if the len of out.Proposals + out.Transactions is not what we expect?

Reply:
it's in the test

Reply:
yes but not in the runtime